tests/vlan/test_vlan.py

```diff
@@ -35,71 +33,163 @@ def cfg_facts(duthosts, rand_one_dut_hostname):
     return duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
 
 
+def compare_network(src_ipprefix, dst_ipprefix):
+    src_network = ipaddress.IPv4Interface(src_ipprefix).network
+    dst_network = ipaddress.IPv4Interface(dst_ipprefix).network
+    return src_network.overlaps(dst_network)
+
+
 @pytest.fixture(scope="module")
-def vlan_intfs_list():
-    return [ { 'vlan_id': vlan, 'ip': '192.168.{}.1/24'.format(vlan) } for vlan in vlan_id_list  ]
+def vlan_intfs_dict(cfg_facts, tbinfo):
+    vlan_intfs_dict = {}
+    for k, v in cfg_facts['VLAN'].items():
+        vlanid = v['vlanid']
+        for addr in cfg_facts['VLAN_INTERFACE']['Vlan'+vlanid]:
+            if addr.find(':') == -1:
+                ip = addr
+                break
+        else:
+            continue
+        logger.info("Original VLAN {}, ip {}".format(vlanid, ip))
+        vlan_intfs_dict[int(vlanid)] = {'ip': ip, 'orig': True}
+    # For t0 topo, will add 2 VLANs for test.
+    # Need to make sure vlan id is unique, and avoid vlan ip network overlapping.
+    # For example, ip prefix is 192.168.0.1/21 for VLAN 1000,
+    # Below ip prefix overlaps with 192.168.0.1/21, and need to skip:
+    # 192.168.0.1/24, 192.168.1.1/24, 192.168.2.1/24, 192.168.3.1/24,
+    # 192.168.4.1/24, 192.168.5.1/24, 192.168.6.1/24, 192.168.7.1/24
+    if tbinfo['topo']['name'] != 't0-56-po2vlan':
+        vlan_cnt = 0
+        for i in xrange(0, 255):
+            vid = 100 + i
+            if vid in vlan_intfs_dict:
+                continue
+            ip = u'192.168.{}.1/24'.format(i)
+            for v in vlan_intfs_dict.values():
+                if compare_network(ip, v['ip']):
+                    break
+            else:
+                logger.info("Add VLAN {}, ip {}".format(vid, ip))
+                vlan_intfs_dict[vid] = {'ip': ip, 'orig': False}
+                vlan_cnt += 1
+            if vlan_cnt >= 2:
+                break
+    return vlan_intfs_dict
```

Comment:
>vlan_intfs_dict [](http://example.com/codeflow?start=11&length=15)

Is it possible that extreme case will never arrive `vlan_cnt >= 2`? Will it be a problem to following code?

Reply:
It is possible that this function can't create new VLAN. 
If t0 topo has already configured 4096 VLAN, then following code can't get any vlanport for test, and will pass the test.
And t0-56-po2vlan topo will not add new VLAN.

Reply:
It is a extreme case that above code could not handle. I suggest add a hard stop here with explicit explanation in code comment..